src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -1099,4 +1099,44 @@ public static void preventIllegalAccessWarnings()
             // ignore
         }
     }
+
+    public static String camelToSnake(String str)
+    {
+        // taken from https://www.geeksforgeeks.org/convert-camel-case-string-to-snake-case-in-java/
+        // Empty String
+        String result = "";
+
+        // Append first character(in lower case)
+        // to result string
+        char c = str.charAt(0);
+        result = result + Character.toLowerCase(c);
+
+        // Traverse the string from
+        // ist index to last index
+        for (int i = 1; i < str.length(); i++)
+        {
+            char ch = str.charAt(i);
+
+            // Check if the character is upper case
+            // then append '_' and such character
+            // (in lower case) to result string
+            if (Character.isUpperCase(ch))
+            {
+                result = result + '_';
+                result
+                = result
+                  + Character.toLowerCase(ch);
+            }
+
+            // If the character is lower case then
+            // add such character into result string
+            else
+            {
+                result = result + ch;
+            }
+        }
+
+        // return the result
+        return result;
+    }
```

Comment:
If you want to trade the string concatenations in loops for an array copy...

```
String camelToSnake(String camel, boolean upper)
{
    StringBuilder stringBuilder = new StringBuilder();
    for (char c : camel.toCharArray())
    {
        char nc = upper ? Character.toUpperCase(c) : Character.toLowerCase(c);
    
        if (Character.isUpperCase(c))
            stringBuilder.append('_').append(nc);
        else
            stringBuilder.append(nc);
    }
    return stringBuilder.toString();
}
```

Reply:
Either way, might be good for it to have at least a couple unit tests.

Reply:
working on the unit tests, your version had a bug =D